### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by stanleySimeon**
# In this milestone I :

_used validations for models and wrote unit tests for models_

  - For the **User** model I make sure that:
    - the `name` isn't blank
    - the `PostsCounter` is an integer and greater than or equal to zero

  - For the **Post** model I make sure that:
    - the `title` isn't blank and doesn't exceed 250 characters
    - the `CommentsCounter` is an integer and greater than or equal to zero
    - the `LikesCounter` is an integer and greater than or equal to zero

  - and I added unit specs for all models, methods, and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    author.increment!(:posts_counter)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  validates :title, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (FrederickMih) [inline app/models/post.rb]: - [Optional] I think is a good practice to do validations before adding methods

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  after_save :update_posts_counter
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  validates :title, length: { maximum: 250 }
```
Comment (FrederickMih) [inline app/models/post.rb]: - [x] Based on the requirements of this project, the title should not be blank
- Hints `validates :title, presence: true`

Let's kindly fix this issue

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @stanleySimeon , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Correct GitHub workflow :heavy_check_mark: 
- Highly descriptive README.md file :heavy_check_mark: 
- PR has a title and summary :+1: 
- No Linter errors :heavy_check_mark: 
- Wrote unit tests for User, Post, Comment and Like model :+1: 

## Required Changes ♻️

- [x] Hi @stanleySimeon, right now, I can't practically check your validations because you didn't configure your routes properly. Kindly set up correct routes in config/routes.rb path and make sure that our app is working properly.

_Check the comments under the review._

## Optional suggestions

- [x] I think is not a good practice to commit database.yml confidential information(user, password) into your repo. It should just be an env var in production, and the default in development.
- **Hint:** Check out [this example](https://gist.github.com/scottrobertson/686abf16e3971696255898097a0bc16e) file



_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (stanleySimeon) [reply app/models/post.rb]: Hi @FrederickMih, kindly, can you be more precise, please?
I'm not sure I understand

File: app/models/post.rb (hunk 2 above)
Comment (stanleySimeon) [reply app/models/post.rb]: Fixed

File: app/models/post.rb (hunk 1 above)
Comment (stanleySimeon) [reply app/models/post.rb]: Fixed

**Review by Lordkaito — APPROVED** [review]
Hi @stanleySimeon, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**